src/lib/features/events/event-service.ts

```diff
@@ -222,6 +229,14 @@ export default class EventService {
     async getEventCreators() {
         return this.eventStore.getEventCreators();
     }
+
+    async getProjectFilterForNonAdmins(userId: number): Promise<IQueryParam[]> {
+        const isRootAdmin = await this.accessReadModel.isRootAdmin(userId);
```

Comment:
This is the main logic. If you are not root admin, add db query filter to not include projects that have null as value.

Reply:
Oh, right. So effectively, this removes any non-project events for non-admins.